pulsar-client-cpp/lib/ProducerConfiguration.cc

```diff
@@ -36,6 +36,25 @@ ProducerConfiguration& ProducerConfiguration::operator=(const ProducerConfigurat
     return *this;
 }
 
+ProducerConfiguration& ProducerConfiguration::setProducerName(const std::string& producerName) {
+    impl_->producerName = Optional<std::string>::of(producerName);
+    return *this;
+}
+
+const std::string& ProducerConfiguration::getProducerName() const {
+    static const std::string emptyString;
+    return impl_->producerName.is_present() ? impl_->producerName.value() : emptyString;
+}
+
+ProducerConfiguration& ProducerConfiguration::setInitialSequenceId(int64_t initialSequenceId) {
+    impl_->initialSequenceId = Optional<int64_t>::of(initialSequenceId);
```

Comment:
Disallow negative value?

Reply:
Just noticed your earlier comment. Should we allow any negative in that case?

Reply:
Yes, initial can also be -1.

Reply:
I was referring to someone setting it to -10 for example. Do we have to reject any value less than -1? Ot it doesn't matter for us?